pylint/checkers/classes.py

```diff
@@ -1086,6 +1092,14 @@ def _check_iter(self, node):
             if not self._is_iterator(infered):
                 self.add_message('non-iterator-returned', node=node)
 
+    def _check_len(self, node):
+        infered = _safe_infer_call_result(node, node)
+        if infered is None or infered is astroid.util.Uninferable:
+            return
+        value = infered.value
```

Comment:
infered might not have a value, no? We can't guarantee it's always a Const value.  Can you do add a test for this?